pkg/alert/alert.go

```diff
@@ -0,0 +1,534 @@
+	}
+	if err := alert.deleteRecoveredTestsFromAlertIndex(testNames); err != nil {
+		return err
+	}
+	return nil
+}
+
+//createAlertMessage creates the alert message
+func createRecoverMessage(recoveredTests map[string]TestDetails) string {
+	sortedKeys := make([]string, 0, len(recoveredTests))
+	for k := range recoveredTests {
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
+
+	writer := &strings.Builder{}
+	table := tablewriter.NewWriter(writer)
+	header := []string{"Test", "Landscape", "Provider", "K8s Ver", "OS"}
+
+	content := make([][]string, 0)
+	row := 0
+	for _, mapKey := range sortedKeys {
+		test := recoveredTests[mapKey]
+		newRow := []string{test.Name,
+			test.Landscape,
+			util.StringDefault(test.Cloudprovider, "-"),
+			util.StringDefault(test.K8sVersion, "-"),
+			util.StringDefault(test.OperatingSystem, "-"),
+		}
+		content = append(content, newRow)
+		row++
+	}
+
+	table.SetHeader(header)
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.AppendBulk(content)
+	table.Render()
+	return writer.String()
+}
+
+func (alert *Alert) deleteRecoveredTestsFromAlertIndex(testNames []string) error {
+	alert.log.V(3).Info("delete recovered tests from elasticsearch tm-alert index")
+	deleteOutdatedAlertDocsPayload := fmt.Sprintf(`{ "query": { "terms" : { "testContext.keyword" : ["%s"] } } }`, strings.Join(testNames,`", "`))
+	if err := alert.elasticRequest("/tm-alert*/_delete_by_query", http.MethodPost, deleteOutdatedAlertDocsPayload, nil); err != nil {
+		return errors.Wrapf(err, "failed to delete recovered elasticsearch test docs")
+	}
+	return nil
+}
+
+//splitSlackMessage split message line wise based on given characters limit
+func splitSlackMessage(message string, charactersLimit int) []string {
+	var messageSplits []string
+	lines := strings.Split(message, "\n")
+	messageSplitItem := ""
+	for _, line := range lines {
+		if len(messageSplitItem)+len(line) < charactersLimit {
+			messageSplitItem += "\n" + line
+		} else {
+			messageSplits = append(messageSplits, messageSplitItem)
+			messageSplitItem = line
+		}
+	}
+	messageSplits = append(messageSplits, messageSplitItem)
+	return messageSplits
+}
+
+//elasticRequest send HTTP request to elasticsearch
+func (alert *Alert) elasticRequest(urlAttributes, httpMethod, payloadFormated string, result interface{}) error {
+	alert.cfg.Elasticsearch.Endpoint.Path = path.Join(urlAttributes)
+	requestUrl := alert.cfg.Elasticsearch.Endpoint.String()
+	payload := strings.NewReader(payloadFormated)
+	alert.log.V(3).Info(fmt.Sprintf("creating HTTP %s request on %s", httpMethod, requestUrl))
+	req, err := http.NewRequest(httpMethod, requestUrl, payload)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create a http request for %s", requestUrl)
+	}
+
+	req.SetBasicAuth(alert.cfg.Elasticsearch.User, alert.cfg.Elasticsearch.Pass)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Cache-Control", "no-cache")
+
+	res, err := http.DefaultClient.Do(req)
```

Comment:
also check if `res.StatusCode` is between 200 and 299. So that we catch http errors codes